engine/src/main/java/org/pentaho/di/job/entries/sendnagiospassivecheck/JobEntrySendNagiosPassiveCheck.java

```diff
@@ -0,0 +1,573 @@
+  }
+
+  /**
+   * @return Returns the password.
+   */
+  public String getPassword() {
+    return password;
+  }
+
+  public int getEncryptionMode() {
+    return encryptionMode;
+  }
+
+  public void setEncryptionMode( int encryptionModein ) {
+    this.encryptionMode = encryptionModein;
+  }
+
+  public int getLevel() {
+    return level;
+  }
+
+  public void setLevel( int levelMode ) {
+    this.level = levelMode;
+  }
+
+  /**
+   * @param message
+   *          The message to set.
+   */
+  public void setMessage( String message ) {
+    this.message = message;
+  }
+
+  /**
+   * @return Returns the comString.
+   */
+  public String getMessage() {
+    return message;
+  }
+
+  /**
+   * @return Returns the port.
+   */
+  public String getPort() {
+    return port;
+  }
+
+  /**
+   * @param port
+   *          The port to set.
+   */
+  public void setPort( String port ) {
+    this.port = port;
+  }
+
+  /**
+   * @param responseTimeOut
+   *          The responseTimeOut to set.
+   */
+  public void setResponseTimeOut( String responseTimeOut ) {
+    this.responseTimeOut = responseTimeOut;
+  }
+
+  /**
+   * @return Returns the responseTimeOut.
+   */
+  public String getResponseTimeOut() {
+    return responseTimeOut;
+  }
+
+  /**
+   * @param connectionTimeOut
+   *          The connectionTimeOut to set.
+   */
+  public void setConnectionTimeOut( String connectionTimeOut ) {
+    this.connectionTimeOut = connectionTimeOut;
+  }
+
+  /**
+   * @return Returns the connectionTimeOut.
+   */
+  public String getConnectionTimeOut() {
+    return connectionTimeOut;
+  }
+
+  public Result execute( Result previousResult, int nr ) {
+    log.logBasic( BaseMessages.getString( PKG, "JobEntrySendNagiosPassiveCheck.Started", serverName ) );
+
+    Result result = previousResult;
+    result.setNrErrors( 1 );
+    result.setResult( false );
+
+    // Target
+    String realServername = environmentSubstitute( serverName );
+    String realPassword = Utils.resolvePassword( variables, password );
+    int realPort = Const.toInt( environmentSubstitute( port ), DEFAULT_PORT );
+    int realResponseTimeOut = Const.toInt( environmentSubstitute( responseTimeOut ), DEFAULT_RESPONSE_TIME_OUT );
+    int realConnectionTimeOut =
+      Const.toInt( environmentSubstitute( connectionTimeOut ), DEFAULT_CONNECTION_TIME_OUT );
+
+    // Sender
+    String realSenderServerName = environmentSubstitute( senderServerName );
+    String realSenderServiceName = environmentSubstitute( senderServiceName );
+
+    try {
+      if ( Utils.isEmpty( realServername ) ) {
+        throw new KettleException( BaseMessages.getString(
+          PKG, "JobSendNagiosPassiveCheck.Error.TargetServerMissing" ) );
+      }
+
+      String realMessageString = environmentSubstitute( message );
+
+      if ( Utils.isEmpty( realMessageString ) ) {
+        throw new KettleException( BaseMessages.getString( PKG, "JobSendNagiosPassiveCheck.Error.MessageMissing" ) );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [KettleException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [544].  Consider adding a catch clause of this specific exception in the outer try statement.